### ISSUE TITLE
False positive `useless-super-delegation` when only passing different arguments

### ISSUE THREAD

**Issue by oskarkk** [pylint-dev/pylint#9226]
### Bug description

```python
`useless-super-delegation` is raised when subclass's `__init__` only changes arguments passed to parent's `__init__`.


class Error:
    def __init__(self, code: str | None = None, message: str | None = None):
        self.code = code
        self.message = message

class IncorrectNumber(Error):
    def __init__(self, *, code: str | None = None, message: str | None = None):
        super().__init__(  #  <--- here
            code=code or "INCORRECT-NUMBER",
            message=message or "Incorrect number",
        )
```

and also 

```
class IncorrectNumber(Error):
    def __init__(self, *, code: str | None = None, message: str | None = None):
        super().__init__(
            code="INCORRECT-NUMBER",
            message="Incorrect number",
        )
```
```


### Configuration

_No response_

### Command used

```shell
pylint pylintrepro.py
```

### Pylint output

```python
************* Module pylintrepro
pylintrepro.py:9:4: W0246: Useless parent or super() delegation in method '__init__' (useless-parent-delegation)

------------------------------------------------------------------
Your code has been rated at 8.75/10 (previous run: 0.00/10, +8.75)
```

### Expected behavior

Not raising a warning

### Pylint version

```shell
pylint 3.0.2
astroid 3.0.1
Python 3.9.18 (main, Oct 13 2023, 18:33:12) 
[GCC 13.2.1 20230801]

and also python 3.11
```
